chapters/data-processing.tex

```diff
@@ -0,0 +1,783 @@
+and its unique identifier will be the combination of household ID and member ID.
+This is a tidy dataset, where
+the \texttt{gender} and \texttt{age} variables are each represented by one column.
+The reshape can be done using \texttt{reshape} in Stata, and \texttt{reshape2} or \texttt{pivot\_longer} in R.
+
+% Polishing tidy data
+One key aspect in this process is to make sure that ID variables are consistent across tables.
+This means that if you want to calculate the share of women members in a household,
+and link it back to the household-level table, 
+you should be able to do so through the household-level row ID.
+Another possible consequence of reshaping to keep in mind is that
+empty observations may be created in the long dataset
+(this is particularly the case in Stata).
+Remember to remove such observations,
+or they may bias aggregated measures --
+counts and proportions in particular.
+
+% Generalizing
+Alternatively, if each row in the untidy data set represents the lower level of observation,
+the same data point for each higher level of observation will be repeated in many rows.
+After you have separated the data into two tables containing the variables observed at each level,
+the table that contains the lower level will already be in a tidy format.
+Then all you need to do is make sure that 
+all rows for the same larger unit have the same value for each variable,
+and drop duplicated observations,
+so that there is only one row per unit.
+The tidying process can get more complex if the instrument includes multiple nested groups.
+In agricultural surveys, it is not uncommon to find household-plot-season-crop level observations,
+as well as different combinations of these level,
+such as household-plot and household-crop-season, in the same dataset.
+That means the steps of identifying the unit of observation of each variable
+and reshaping the separated tables need to be repeated multiple times.
+In these case, it is particularly important to be mindful of cross-table identifiers,
+as they are often not the same as wide variables suffixes.
+
+% Wrap-up
+Tidying datasets can be a complex process.
+However, the more nested groups a dataset includes,
+the more efficient it is to deal with tidy data as compared to untidy.
+Cleaning and constructing wide datasets, in particular,
+is a repetitive and error-prone process.
+The next step of data cleaning, data quality monitoring,
+may involve comparing different units of observation.
+Aggregating sub-units to compare to a higher unit is much easier with tidy data,
+which is why we suggest getting your data into a tidy format as early as possible in your workflow. 
+In addition to that, in the case of survey data,
+tidying datasets will guarantee a one-to-one correspondence
+between questions in the questionnaire and columns in the data.
+Preparing the data for analysis, the last task in this chapter, 
+is much simpler when that is the case. 
+ 
```

Comment:
Here are my overall comments on the tidy content:
- It is an unpolished diamond. All the content is there, and the content is good. But some improvements are needed.
- The main things is that we do not have a clear division on where we introduce vocabulary, intuition/justification and how to apply it. We keep talking about the justification long into how to apply it. I think it should be in this order:
  - Vocabulary: 
    - (table, data point, varaible unit of obs etc.)
  - Unique identifiers
  - Wide vs. Tidy - rename the subsection "_What makes data tidy_" to "_Working with data wide or tidy_"
    - Talk about most data has nested unit of observations. That is the core starting point of tidy.
    - Explain that wide is one table with many variables age_1, age_2 etc., while tidy is many tables but each variable exists only once, where an exact linkage between the tables.
  - Tools and practices working with tidy data
    - we have this sub-section already, but half of it is still comparing wide vs tidy data. By this point we should have made our argument for tidy and only focus on how to work with tidy.
- Through the chapter multiple terms are used to mean the same thing, which will be confusing to a reader not already a data expert. I am thinking especially at sub-unit observation vs. lower lever observation. We need to exaplin one of them in the beginning and then stick to that one through the chapter.
- Finally, maybe it is coming later, in construct section, but -- if it can be worked in somehow -- I think we should already here give the reader the intuition that you are not meant to reshape and merge tidy data back to wide. Even if your regressions are on the highest level unit of obs, you should not do that. You should instead aggregate or in other way transform your sub-units of observations to the parent units with sums and proportion, and then merge to the parent table, and that is still tidy.

Reply:
I see what you are saying now. Some content I think should go into the "wide vs. tidy" subsection:
- Tidy datasets are all alike but every messy dataset is messy in its own way
- Though every messy dataset is messy in its own way, survey datasets are messy in a similar way
- Survey data that contains nested units is typically imported from survey platforms as a "wide" dataset
- This is not because this is the most efficient way to manipulate datasets, but because it's the  most efficient way to put different observations into the same table
- But it is a widespread practice to handle wide data, mainly because of legacy (need a better word here)
- The format we recommend, "tidy" format, is different. It's used because it's the most efficient way to manipulate data, particularly when sub-units are present and need to be aggregated and combined.
- In this format, a wide dataset with different units of observation is broken into several datasets, each of them containing only the variables observed at a given unit of observation
- So household-level variables are stored in one dataset, and household-member variables are stored in another.

Reply:
instead of legacy -> old customs, old practices etc.

Reply:
Here is one example where I think you need to put wide in tidy in contrast to each other to explain your point. 

https://github.com/worldbank/d4di/blob/1029ad47f6590c62a4c27ed9034efb43c8d5bfa2/chapters/data-processing.tex#L138-L141

I can easily see someone who only works with wide data because everyone around them does that and no one has ever questioned it say: "_Sure, I also only have one column per variable, and they are called age_1, age_2, age_3 etc._" Once you understand tidy you see how that sentence is incorrect, but we should assume that the audience of this book does not understand tidy yet. In Stata each column is called a variable, so in that aspect it is always a one-to-one relationship between column and variable, making the sentence "_each column represents one variable_" sound unnecessary. Does this make sense?

Reply:
Add the wide example in the very beginning of the section, as one common example of untidy data

Reply:
done in 704d2b2